Utilities/Mergers/src/MergerBuilder.cxx

```diff
@@ -66,20 +69,20 @@ framework::DataProcessorSpec MergerBuilder::buildSpec()
   merger.inputs = mInputSpecs;
 
   merger.outputs.push_back(mOutputSpec);
-  if (mOutputSpec.origin == header::gDataOriginInvalid || mOutputSpec.description == header::gDataDescriptionInvalid) {
+  framework::DataAllocator::SubSpecificationType subSpec;
```

Comment:
Could you initialize it with ` = DataSpecUtils::getOptionalSubSpec(mOutputSpec).value();`?

Reply:
Ok, I guess only for the case in which the mOutputSpec contains a value, though.

Reply:
Now Mergers are not expected to work with wildcards, but indeed, the interface does not defend itself from that. You can add something like this to `std::string MergerInfrastructureBuilder::validateConfig()`. If not, I can put it in a separate PR.
```
  if (mInputs.empty()) {
    error += preamble + "no inputs specified\n";
  }
  for (const auto& input : mInputs) {
    if (std::holds_alternative<ConcreteDataMatcher>(input.matcher) == false) {
      error += preamble + "one or more inputs are not concrete\n";
      break;
    }
  }
  if (DataSpecUtils::validate(mOutputSpec) == false) {
    error += preamble + "invalid output\n";
  }
  if (std::holds_alternative<ConcreteDataMatcher>(mOutputSpec.matcher) == false) {
    error += preamble + "the output is not concrete\n";
  }
```